src/rooms/RestartRoom.ts

```diff
@@ -11,11 +11,36 @@ export class RestartRoom extends Room<RestartRoomState> {
   maxClients = 8;
   LOBBY_CHANNEL = "$restartlobby";
 
+  //Constructor for disposeTimer
+  private disposeTimer: NodeJS.Timeout | null = null;
+
+  constructor() {
+    super();
+    this.startDisposeTimer = this.startDisposeTimer.bind(this);
+  }
+
+  private startDisposeTimer() {
+    this.disposeTimer = setTimeout(() => {
```

Comment:
This will not have the intended behavior as this will only check the number of clients once after 1 min 10 sec the room get initialized. Instead, the ideal behavior should be that after the second last client disconnects, it waits for 10 sec and if still there is <=1 client, it disposes the room. 
Edge cases like another client joined after the second last client disconnects or only one client joined since the room init should be separately taken care of. Think of how Google Meet works.

Reply:
This might help: https://discuss.colyseus.io/topic/504/destroy-a-room-after-a-set-duration-if-no-one-joined-it-sovled